PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java

```diff
@@ -0,0 +1,249 @@
+package org.prebid.mobile;
+
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.prebid.mobile.testutils.BaseSetup;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import java.util.ArrayList;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertNotNull;
+import static junit.framework.Assert.assertNull;
+import static junit.framework.Assert.assertTrue;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(sdk = BaseSetup.testSDK, manifest=Config.NONE)
+public class NativeAdUnitTest {
+    static final String PBS_CONFIG_ID_NATIVE_APPNEXUS = "1f85e687-b45f-4649-a4d5-65f74f2ede8e";
+
+    @Test
+    public void testNativeAdUnitCreation() throws Exception {
+        NativeAdUnit nativeUnit = new NativeAdUnit(PBS_CONFIG_ID_NATIVE_APPNEXUS);
+        assertNotNull(nativeUnit);
+        assertEquals(PBS_CONFIG_ID_NATIVE_APPNEXUS, FieldUtils.readField(nativeUnit, "configId", true));
+        assertEquals(AdType.NATIVE, FieldUtils.readField(nativeUnit, "adType", true));
+    }
+
+    @Test
+    public void testNativeAdContextType() {
+        NativeAdUnit nativeUnit = new NativeAdUnit(PBS_CONFIG_ID_NATIVE_APPNEXUS);
+        assertNotNull(nativeUnit);
+        assertNull(nativeUnit.params.getContextType());
+        nativeUnit.setContextType(NativeAdUnit.CONTEXT_TYPE.CONTENT_CENTRIC);
+        assertNotNull(nativeUnit.params.getContextType());
+        assertEquals(nativeUnit.params.getContextType(), NativeAdUnit.CONTEXT_TYPE.CONTENT_CENTRIC);
+        nativeUnit.setContextType(NativeAdUnit.CONTEXT_TYPE.SOCIAL_CENTRIC);
+        assertEquals(nativeUnit.params.getContextType(), NativeAdUnit.CONTEXT_TYPE.SOCIAL_CENTRIC);
+        nativeUnit.setContextType(NativeAdUnit.CONTEXT_TYPE.PRODUCT);
+        assertEquals(nativeUnit.params.getContextType(), NativeAdUnit.CONTEXT_TYPE.PRODUCT);
+        nativeUnit.setContextType(NativeAdUnit.CONTEXT_TYPE.CUSTOM);
+        assertEquals(nativeUnit.params.getContextType(), NativeAdUnit.CONTEXT_TYPE.CUSTOM);
```

Comment:
Try to set a custom id then test the returned value.

Reply:
Done